ms-engine/src/main/java/net/trajano/ms/engine/SpringJaxRsHandler.java

```diff
@@ -318,58 +316,57 @@ public void handle(final RoutingContext context) {
         if (isMultipartExpected(request)) {
             context.request().setExpectMultipart(true);
         }
-        try (final VertxHttpResponse response = new VertxHttpResponse(context)) {
-            context.vertx().executeBlocking(
-                future -> {
-                    if (LOG.isDebugEnabled()) {
-                        LOG.debug("{} {}", context.request().method(), serverRequest.absoluteURI());
-                    }
-                    ThreadLocalResteasyProviderFactory.push(providerFactory);
-                    ResteasyProviderFactory.pushContext(RoutingContext.class, context);
-                    ResteasyProviderFactory.pushContext(Vertx.class, context.vertx());
-                    ResteasyProviderFactory.pushContext(Client.class, client);
-                    ResteasyProviderFactory.pushContext(HttpHeaders.class, request.getHttpHeaders());
-
-                    try {
-                        dispatcher.invokePropagateNotFound(request,
-                            response);
-                        future.complete();
-                    } finally {
-                        ResteasyProviderFactory.clearContextData();
-                        ThreadLocalResteasyProviderFactory.pop();
-                    }
-                }, false,
-                res -> {
-                    if (res.failed()) {
-                        final Throwable wae = res.cause();
-                        if (wae instanceof NotFoundException) {
-                            LOG.debug("uri={} was not found", serverRequest.uri());
-                            context.next();
-                        } else {
-                            context.fail(wae);
-                        }
+        final VertxHttpResponse response = new VertxHttpResponse(context);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "VertxHttpResponse" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)

Reply:
DO NOT CLOSE THIS at least not on the same thread.  Or ignore the close event.